src/openms/include/OpenMS/ANALYSIS/OPENSWATH/DIAHelper.h

```diff
@@ -128,18 +128,31 @@ namespace OpenMS
                               std::vector<std::pair<double, double> >& modmass, //!< [out]
                               double charge = 1.);
 
-    /// add negative pre-isotope weights to spectrum
+    /// add (potentially negative) pre-isotope weights to spectrum
     OPENMS_DLLAPI void addPreisotopeWeights(const std::vector<double>& firstIsotopeMasses,
                               std::vector<std::pair<double, double> >& isotopeSpec, // output
                               UInt nrpeaks = 2, //nr of pre-isotope peaks
                               double preIsotopePeaksWeight = -0.5, // weight of pre-isotope peaks
                               double mannmass = 1.000482, //
                               double charge = 1.);
 
+    /// add negative pre-isotope weights to spectrum
+    OPENMS_DLLAPI void addPreisotopeWeights(double mz,
+                                            std::vector<std::pair<double, double> >& isotopeSpec, // output
+                                            UInt nrpeaks = 2, //nr of pre-isotope peaks
+                                            double preIsotopePeaksWeight = -0.5, // weight of pre-isotope peaks
+                                            double mannmass = 1.000482, //
+                                            double charge = 1.);
+
     /// given an experimental spectrum add isotope pattern.
     OPENMS_DLLAPI void addIsotopes2Spec(const std::vector<std::pair<double, double> >& spec,
                           std::vector<std::pair<double, double> >& isotopeMasses, //[out]
-                          double charge = 1.);
+                          Size nr_isotopes, double charge = 1.);
+
+    /// given a peak of experimental mz and intensity, add isotope pattern to a "spectrum".
+    OPENMS_DLLAPI void addIsotopes2Spec(double mz, double ity,
```

Comment:
is there a better name that is more expressive: 
addIsotopesOfASinglePeak2Spectrum (comment if it is according to averagine?)

Reply:
maybe even addTheoIsotopes... -> to emphasise that it is the theoretical isotope pattern? I was actually confused in the beginning.